brave/brave-core
# Removes `std::function<>` from further `Ledger` flows.

Resolves https://github.com/brave/brave-browser/issues/24200.

Removes `std::function<>` from:
- the `Ledger::CreateWallet()` flow
- the `Ledger::GetRewardsParameters()` flow
- the `Ledger::SetPublisherExclude()` flow
- the `Ledger::RestorePublishers()` flow
- the `Ledger::FetchPromotions()` flow
- the `Ledger::ClaimPromotion()` flow

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

- test if creating a Rewards wallet works on all platforms
- make sure Rewards parameters are fetched/displayed correctly on all platforms (search for `https://api.rewards.bravesoftware.com/v1/parameters` in logs):
  - BAT to USD exchange rate (e.g. on the tipping panel)
  - auto-contribute choices in `Auto-Contribute Settings` on `brave://rewards`
  - monthly tip choices on the monthly tip form
  - one-time tip choices on the one-time tip form
  - payout status
- check if excluding and restoring publishers work as expected on all platforms
- check if fetching and claiming promotions work as expected on all platforms
